src/lib/generators/code-snippet-generators.ts

```diff
@@ -2,30 +2,55 @@ import snakeCase from 'lodash.snakecase';
 import camelCase from 'lodash.camelcase';
 import kebapCase from 'lodash.kebabcase';
 import { SvgDefinition } from '../converters/shared.converter';
+import { MultiFileConvertionOptions, SingleFileConvertionOptions } from '../options/convertion-options';
 
 export enum Delimiter {
   CAMEL = 'CAMEL',
   KEBAB = 'KEBAB',
   SNAKE = 'SNAKE'
 }
 
-export const generateInterfaceDefinition = (interfaceName: string, typeName: string) => {
-  return `export interface ${interfaceName}{
-        name: ${typeName};
+export const generateInterfaceDefinition = (
+  convertionOptions: MultiFileConvertionOptions | SingleFileConvertionOptions
+) => {
+  const iconNameType =
+    convertionOptions.generateType || convertionOptions.generateTypeObject ? convertionOptions.typeName : 'string';
+  return `export interface ${convertionOptions.interfaceName}{
+        name: ${iconNameType};
         data: string;}`;
 };
 
-export const generateTypeDefinition = (typeName: string, svgDefinitions: SvgDefinition[]): string => {
-  const typesDelimitor = ' | ';
-  const typeHeader = `
+export const generateTypeDefinition = (
+  convertionOptions: MultiFileConvertionOptions | SingleFileConvertionOptions,
+  svgDefinitions: SvgDefinition[]
+): string => {
+  let typesDefinition = `
     /* ⚠️ Do not edit this file - this file is generated by svg-to-ts*/
     
-    export type ${typeName} = 
   `;
-  const types = svgDefinitions.map(({ typeName }, index) =>
-    index === svgDefinitions.length - 1 ? `'${typeName}';` : `'${typeName}'${typesDelimitor}`
-  );
-  return typeHeader + types.join('');
+
+  if (convertionOptions.generateType || convertionOptions.generateTypeObject) {
```

Comment:
I would remove the or check - I would say that the user should be able to generate either a type or an object or both.